src/annotator/plugin/test/pdf-test.js

```diff
@@ -0,0 +1,121 @@
+import PDF, { $imports } from '../pdf';
+
+import FakePDFViewerApplication from '../../anchoring/test/fake-pdf-viewer-application';
+
+function delay(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+describe('annotator/plugin/pdf', () => {
+  let container;
+  let fakeAnnotator;
+  let fakePDFViewerApplication;
+  let pdfPlugin;
+
+  function createPDFPlugin() {
+    const plugin = new PDF(document.body);
+    plugin.annotator = fakeAnnotator;
+    plugin.pluginInit();
+    return plugin;
+  }
+
+  beforeEach(() => {
+    // Setup fake PDF.js viewer.
+    container = document.createElement('div');
+    document.body.appendChild(container);
+    fakePDFViewerApplication = new FakePDFViewerApplication({
+      container,
+      content: ['First page', 'Second page'],
+    });
+    fakePDFViewerApplication.pdfViewer.setCurrentPage(0);
+    window.PDFViewerApplication = fakePDFViewerApplication;
+
+    fakeAnnotator = {
+      anchor: sinon.stub(),
+      anchors: [],
+      anchoring: null,
+    };
+
+    $imports.$mock({
+      // Disable debouncing of updates.
+      'lodash.debounce': callback => callback,
+    });
+  });
+
+  afterEach(() => {
+    container.remove();
+    pdfPlugin?.destroy();
+    delete window.PDFViewerApplication;
+
+    $imports.$restore();
+  });
+
+  describe('#constructor', () => {
+    it('adds CSS classes to override PDF.js styles', () => {
```

Comment:
Is _that_ the purpose of `has-transparent-text-layer?!

Reply:
Yes. The `pdfjs-overrides.scss` source file has more details. Perhaps we should change the class name to `hypothesis-active` or something? That's outside the scope of this PR though.